### ISSUE TITLE
Where is an upgraded client?

### ISSUE THREAD

**Issue by tomtruitt** [bitpay/bitcore-wallet-service#584]
wallet create Alexandra 2-3
[info] Generating new keys
[error] <ClientError:UPGRADE_NEEDED BWC clients < 1.2 are no longer supported.>
ClientError {
  code: 'UPGRADE_NEEDED',
  message: 'BWC clients < 1.2 are no longer supported.' }

**Comment by dabura667** [comment]
https://github.com/bitpay/bitcore-wallet-client

**Comment by tomtruitt** [comment]
That's the client library... In the readme it point to a CLI client. Is there no command line client that's supported any longer?

**Comment by tomtruitt** [comment]
The client in the readme is listed as https://github.com/bitpay/bitcore-wallet

**Comment by dabura667** [comment]
current CLI at the link you provided requires 2.3.1 of BWC, so you should not see error message about <1.2

Upgrade your bitcore-wallet

**Comment by tomtruitt** [comment]
Yes you'll see from the title of my issue I don't know how. I cloned bitcore-wallet and yesterday and received this error.

**Comment by dabura667** [comment]
did you perform `npm install` in the cloned directory? nodejs packages need to run install to install the required dependencies... perhaps you have an old version of BWC on your system.

**Comment by tomtruitt** [comment]
To be honest I'm not the one who installed it. I guess I shouldn't assume things. Let me check thank you.

**Comment by tomtruitt** [comment]
I'm confused the version at the link i sent is clearly labelled 0.6.1

**Comment by dabura667** [comment]
The error you sent 'BWC clients < 1.2 are no longer supported.' means that your instance of bitcore-wallet is using an old version of bitcore-wallet-client. If you look at the most recent version of bitcore-wallet, the dependencies are stated as follows. (Check package.json)

```javascript
  "dependencies": {
    "bitcore-wallet-client": "git://github.com/bitpay/bitcore-wallet-client#41abd4fd6365db083cda9d305978121a03e86d6e",
    "commander": "^2.6.0",
    "lodash": "^3.3.1",
    "moment": "^2.9.0",
    "npmlog": "^1.2.0",
    "qr-image": "^3.1.0",
    "read": "^1.0.5",
    "sjcl": "^1.0.2"
  },
```

You can view the source for the commit noted at this link:
https://github.com/bitpay/bitcore-wallet-client/tree/41abd4fd6365db083cda9d305978121a03e86d6e

If you check package.json for bitcore-wallet-client at this commit, it says 

```javascript
"version": "2.3.1",
```

So just to test it out, I just ran:

```bash
$ git clone https://github.com/bitpay/bitcore-wallet
$ cd bitcore-wallet
$ npm install
$ grep "\"version\":" node_modules/bitcore-wallet-client/package.json
>  "version": "2.3.1",
```

**Comment by tomtruitt** [comment]
yes my version of BWC is also 2.3.1

you'll notice the error says "BWC clients" under version 1.2 are no longer supported... the command line client https://github.com/bitpay/bitcore-wallet is listed as 0.6.2

**Comment by dabura667** [comment]
BWC client is referring to bitcore-wallet-client. I know this for a fact because I have been following all these repos for over a year and I know why this error was created.

bitcore-wallet-client version 1.2 included code that breaks compatibility with <1.2 versions.

The error message has nothing to do with bitcore-wallet, even if you made your own CLI wallet software, if you use <1.2 bitcore-wallet-client, you will get this message.

**Comment by tomtruitt** [comment]
Well according to what you are saying what is happening should not be. The language looks pretty clear. This error should be rewritten at the least because it says bwc client. Regardless clone both repos yesterday 2 days ago on a brand new digital ocean droplet with node and npm installed and bwc reports the latest version

**Comment by tomtruitt** [comment]
Are you able to you the command line client bitcore-wallet?

**Comment by dabura667** [comment]
```bash
$ git clone https://github.com/bitpay/bitcore-wallet
$ cd bitcore-wallet
$ npm install
$ grep "\"version\":" node_modules/bitcore-wallet-client/package.json
>  "version": "2.3.1",
```

Try this. I literally did this just now and got no error.

**Comment by tomtruitt** [comment]
I have done that without errors too. However as my op stated when attempting the `wallet create` command I receive the version error listed.

**Comment by tomtruitt** [comment]
Also your string appears to be looking for something that was already there in bwc repo before the bitcore-wallet install.

**Comment by dabura667** [comment]
Change 

npm install

to

npm install -g .

Don't forget the period.

The wallet command is from the globally installed bitcore-wallet, you are not using the bitcore-wallet you cloned.

**Comment by tomtruitt** [comment]
Ok that makes sense I'll try that thank you!

**Comment by esseak** [comment]
bitcore-wallet-client in bitcore-wallet's dependencies version too low.
If you not use it globally just use in your project try following:

- rm -rf /YOURPROJECT_PATH/node_modules/bitcore-wallet-client #delete low version dependency
- npm install bitcore-wallet-client --save #reinstall bitcore-wallet-client ,v4.4.0 works for me
